### PR TITLE
Corrected DigitalOcean Provider

### PR DESCRIPTION
**Pull request by TheDanielDoyle**
* Corrected multiple issues with DigitalOcean provider.
* Changed DNSRecordCollection to handle most of the Data and Id changes. This mean the hydrater can be simplified.

### CODE REVIEW

File: src/DDNSUpdate.Tests/Application/DDNSUpdateInvokerTests.cs [hunk 1]
```diff
@@ -26,21 +27,21 @@ public async Task SingleExceptionIsCaught()
 
             A.CallTo(() => configurationValidator.ValidateAsync(A<CancellationToken>.Ignored)).Returns(Result.Ok());
 
-            A.CallTo(() => externalAddressClient.GetAsync(A<CancellationToken>.Ignored))
-                .Returns(Task.FromResult(Result.Ok<IExternalAddressResponse>(new ExternalAddressResponse(default))));
+            A.CallTo(() => externalAddressClient.GetAsync(A<CancellationToken>.Ignored)).Returns(Task.FromResult(Result.Ok<IExternalAddressResponse>(new ExternalAddressResponse(default))));
```
Comment (grab-a-byte) [inline src/DDNSUpdate.Tests/Application/DDNSUpdateInvokerTests.cs]: Shouldnt need Task.FromResult in the returns

File: src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs [hunk 2]
```diff
@@ -26,18 +26,20 @@ public DigitalOceanClient(HttpClient httpClient)
...
+            HttpResponseMessage response = await httpRequest.PostJsonAsync(request, cancellation);
+            bool created = (int)response.StatusCode == (int)HttpStatusCode.Created;
+            return Result.OkIf(created, $"Unable to create DNS record for {request.Name}.");
         }
 
         public async Task<Result<DigitalOceanGetDomainRecordsResponse>> GetDNSRecordsAsync(DigitalOceanDomain domain, string token, CancellationToken cancellation)
         {
             string path = string.Format(_getDNSRecordFormat, domain.Name);
-            HttpResponseMessage response = await BuildRequest(token, path).GetAsync(cancellation);
+            IFlurlRequest httpRequest = BuildRequest(token, path);
```
Comment (grab-a-byte) [inline src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs:41]: Is there a reason this is split up? Could still be 1-lined

Comment (TheDanielDoyle) [reply src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs:41]: It's confusing to return a HttpResponseMessage from BuildRequest.
I doubt people would notice GetAsync on the end.

File: src/DDNSUpdate.Tests/Application/DDNSUpdateInvokerTests.cs (hunk 1 above)
Comment (TheDanielDoyle) [reply src/DDNSUpdate.Tests/Application/DDNSUpdateInvokerTests.cs]: Done.

File: src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs (hunk 2 above)
Comment (grab-a-byte) [reply src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs:41]: Eh sure

**Review by grab-a-byte — APPROVED** [review]

**Pull request merged**